### PR TITLE
registry: add GetLatestSchemaInfo function

### PR DESCRIPTION
**Pull request by mcgrawia**
RE issue: #53

### CODE REVIEW

File: registry/client.go [hunk 1]
```diff
@@ -62,6 +65,31 @@ type credentials struct {
...
+		ID: s.ID,
+		Version: s.Version,
+		Subject: s.Subject,
+	}
+	schema, err := avro.Parse(s.Schema)
+	info.Schema = schema
+	return info, err
+}
+
+type SchemaInfo struct {
```
Comment (nrwiersma) [inline registry/client.go:86]: This is missing a comment.

File: registry/client.go [hunk 2]
```diff
@@ -62,6 +65,31 @@ type credentials struct {
 	password string
 }
 
+type schemaInfoPayload struct {
+	Schema  string `json:"schema"`
+	ID      int    `json:"id"`
+	Version int    `json:"version"`
+	Subject string `json:"subject"`
```
Comment (nrwiersma) [inline registry/client.go:72]: We dont need to spend resources unmarshalling this, it was passed into the function.

File: registry/client.go [hunk 3]
```diff
@@ -62,6 +65,31 @@ type credentials struct {
...
+func (s *schemaInfoPayload) Parse() (SchemaInfo, error) {
+	info := SchemaInfo{
+		ID: s.ID,
+		Version: s.Version,
+		Subject: s.Subject,
+	}
+	schema, err := avro.Parse(s.Schema)
+	info.Schema = schema
+	return info, err
+}
```
Comment (nrwiersma) [inline registry/client.go:84]:
```suggestion
func (s *schemaInfoPayload) Parse() (info SchemaInfo, err error) {
	info = SchemaInfo{
		ID: s.ID,
		Version: s.Version,
		Subject: s.Subject,
	}
	info.Schema, err = avro.Parse(s.Schema)
	return info, err
}
```

File: registry/client.go [hunk 4]
```diff
@@ -62,6 +65,31 @@ type credentials struct {
...
+
+type SchemaInfo struct {
+	Schema  avro.Schema
+	ID      int
+	Version int
+	Subject string
```
Comment (nrwiersma) [inline registry/client.go:90]: Dont think we need this. I dont particularly like passing info back that was given to us.

File: registry/client.go (hunk 3 above)
Comment (mcgrawia) [reply registry/client.go:84]: I actually had something like this originally, but I get a compile error saying that `err` is not defined. The following works though:
```go
func (s *schemaInfoPayload) Parse() (SchemaInfo, error) {
	var err error
	info := SchemaInfo{
		ID: s.ID,
		Version: s.Version,
	}
	info.Schema, err = avro.Parse(s.Schema)
	return info, err
}
```

Would you prefer this?

Comment (mcgrawia) [reply registry/client.go:84]: oh whoops, did not see you named the return types. That fixes it. Thanks!

**Comment by mcgrawia** [comment]
ok @nrwiersma I think I've fixed everything. Let me know if you have other change requests

File: registry/client.go [hunk 5]
```diff
@@ -62,6 +65,30 @@ type credentials struct {
...
 
+type schemaInfoPayload struct {
+	Schema  string `json:"schema"`
+	ID      int    `json:"id"`
+	Version int    `json:"version"`
+}
+
+// Parse converts the string schema registry response into a
+// SchemaInfo object with an avro.Schema schema
+func (s *schemaInfoPayload) parse() (info SchemaInfo, err error) {
```
Comment (nrwiersma) [inline registry/client.go:76]: There is no need to privatise this function. IMO it gives the impression it is used internally by `schemaInfoPayload` and is not meant for use by other functions. The struct is already private.

File: .gitignore [hunk 6]
```diff
@@ -0,0 +1 @@
+.idea
```
[the remark this answers is not in the document]
Comment (mcgrawia) [reply .gitignore:1]: yup that makes sense 👍

File: registry/client.go (hunk 5 above)
Comment (mcgrawia) [reply registry/client.go:76]: ok that makes senes. fixed!

**Comment by nrwiersma** [comment]
Fixes #53

**Comment by nrwiersma** [comment]
Thank you for the contribution.

**Pull request merged**